src/main/java/org/kairosdb/core/aggregator/FilterAggregator.java

```diff
@@ -19,121 +19,126 @@
-			return ret;
-		}
-
-		private void moveCurrentDataPoint()
-		{
-			if (hasNextInternal())
-				currentDataPoint = nextInternal();
-			else
-				currentDataPoint = null;
-		}
-	}
+public class FilterAggregator implements Aggregator {
+    public enum FilterOperation {
+        LTE, LT, GTE, GT, EQUAL, NE
+    }
+
+    public FilterAggregator() {
+        m_threshold = 0.0;
+    }
+
+    public FilterAggregator(FilterOperation filterop, Object threshold) {
+        m_filterop = filterop;
+        m_threshold = threshold;
+    }
+
+    @FeatureProperty(
+            name = "filter_op",
+            label = "Filter operation",
+            description = "The operation performed for each data point.",
+            type = "enum",
+            options = {"lte", "lt", "gte", "gt", "equal", "ne"},
+            default_value = "equal"
+    )
+    private FilterOperation m_filterop;
+
+    @FeatureProperty(
+            label = "Threshold",
+            description = "The value the operation is performed on. If the operation is lt, then a null data point is returned if the data point is less than the threshold."
+    )
+    private Object m_threshold;
+
+    /**
+     * Sets filter operation to apply to data points. Values can be LTE, LE, GTE, GT, NE, or EQUAL.
+     *
+     * @param filterop
+     */
+    public void setFilterOp(FilterOperation filterop) {
+        m_filterop = filterop;
+    }
+
+    public void setThreshold(Object threshold) {
+        m_threshold = threshold;
+    }
+
+    public DataPointGroup aggregate(DataPointGroup dataPointGroup) {
+        return new FilterDataPointAggregator(dataPointGroup);
+    }
+
+    public boolean canAggregate(String groupType) {
+        return true;
+    }
+
+    public String getAggregatedGroupType(String groupType) {
+        return groupType;
+    }
+
+    private class FilterDataPointAggregator extends AggregatedDataPointGroupWrapper {
+        public FilterDataPointAggregator(DataPointGroup innerDataPointGroup) {
+            super(innerDataPointGroup);
+        }
+
+
+        private <T> boolean filterOut(FilterOperation op, Comparable<T> value, T threshold) {
+            switch (op) {
+                case LTE:
+                    return value.compareTo(threshold) <= 0;
+                case LT:
+                    return value.compareTo(threshold) < 0;
+                case GTE:
+                    return value.compareTo(threshold) >= 0;
+                case GT:
+                    return value.compareTo(threshold) > 0;
+                case EQUAL:
+                    return value.compareTo(threshold) == 0;
+                case NE:
+                    return value.compareTo(threshold) != 0;
+                default:
+                    return false;
+
+            }
+        }
+
+        public boolean hasNext() {
+            boolean foundValidDp = false;
+            while (!foundValidDp && currentDataPoint != null) {
+                boolean filterOutValue = false;
```

Comment:
It's probably a better to default `filterOutValue` to true instead of false.

As it currently stands, the filter condition will only be checked if the threshold is numeric (actually, if it's a double) *or* if the the data point value is a string.

If the threshold is a string, and the data point value is numeric, then then the data point will never be filtered out. I think that in case of a mismatch between the type of the threshold and the type of the data point value we always want to filter out the data point.